### ISSUE TITLE
Missing type map configuration or unsupported mapping error (not working after update to 4.0.4)

### ISSUE THREAD

**Issue by arturoribes** [AutoMapper/AutoMapper#850]
The following code worked in version 3.3.1 but not after an update to 4.0.4. I realised that if I remove the constructor from TestEntityViewModel, the test passes, otherwise, it gives an error.

```c#
public class TestEntity
{
    public int Id { get; set; }
}

public class TestEntityViewModel
{
    public TestEntityViewModel()
    {

    }

    public TestEntityViewModel(IEnumerable<SelectListItem> testList)
    {
        TestList = testList;
    }

    public int Id { get; set; }
    public virtual IEnumerable<SelectListItem> TestList { get; set; }
}


[TestClass]
public class UnitTest1
{
    [TestMethod]
    public void TestMethod1()
    {
        Mapper.CreateMap<TestEntity, TestEntityViewModel>()
            .ForMember(m => m.TestList, opt => opt.Ignore());

        var entity = new TestEntity();

        var viewModel = Mapper.Map<TestEntityViewModel>(entity);


        Assert.IsNotNull(viewModel);
    }
}
```

The error is the following:

```
Test method WebUI.Tests.UnitTest1.TestMethod1 threw exception: 

AutoMapper.AutoMapperMappingException: Missing type map configuration or unsupported mapping.



Mapping types:

TestEntity -> IEnumerable`1

WebUI.Tests.TestEntity -> System.Collections.Generic.IEnumerable`1[[System.Web.Mvc.SelectListItem, System.Web.Mvc, Version=5.2.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]]



Destination path:

TestEntityViewModel
```

**Comment by lbargaoanu** [comment]
The same as #847 I suppose.

**Comment by lbargaoanu** [comment]
Probably fixed by #816.

**Comment by jbogard** [comment]
Dupe of #847
